sdk/identity/azure-identity/azure/identity/_credentials/authorization_code.py

```diff
@@ -37,9 +37,9 @@ def __init__(
             client_id: str,
             authorization_code: str,
             redirect_uri: str,
-            **kwargs
+            **kwargs: Any
     ) -> None:
-        self._authorization_code = authorization_code  # type: Optional[str]
+        self._authorization_code: Optional[str] = authorization_code
```

Comment:
Looks like it should just be `str`?

Reply:
Unfortunately, authorization_code could be None. It is single-use and once it is used, it needs to be set to None.